README.md

```diff
@@ -1,61 +1,75 @@
+# Scheduling APIs
+
+This document outlines the motivation for working on various scheduling
+APIs<sup>1</sup>, discusses some of the problems that apps and userspace
+schedulers face when writing scheduling code, and links to [various
+proposals](#scheduling-apis-and-status) we are working on in this space.
+
+(<sup>1</sup>The scope of this work was previously restricted to main-thread
+scheduling, and while main-thread scheduling remains the primary focus, the
+repository and some accompanying text has been renamed to "scheduling-apis" to
+reflect the inclusion of APIs like
+[`scheduler.postTask()`](explainers/prioritized-post-task.md) on workers.)
+
+## Motivation: Main-thread Contention
+
+Applications may experience main-thread contention at various points in their
+execution, e.g. during page load or as a result of user interaction. This
+contention can negatively affect user experience in terms of responsiveness and
+latency. For example, a busy main thread can prevent the UA from servicing
+input, leading to poor responsiveness. Similarly, tasks (e.g. fetch
+completions, rendering, etc.) can experience large queuing durations during
+times of contention, which increases task latency and can result in degraded
+quality of experience.
+
+Consider a "search-as-you-type" application. This app needs to be responsive to
+user input, i.e. users typing in the search-box. At the same time, any
+animations on the page must be rendered smoothly, and the work for fetching and
+preparing search results and updating the page must also progress quickly.
+There are a lot of different deadlines to meet for the app developer. It is
+easy for any long running script work to hold up the main thread and cause
 responsiveness issues for typing, rendering animations, or updating search
 results.
 
-(**TODO**: Add a use cases doc with more concrete examples.)
+Another example pinch-zooming in a map application. The app needs to
+continuously respond to the input, update the rendering, and potentially fetch
+new content to be displayed. Similar to the search-as-you-type example, long
+running script work could block other tasks, making the application feel laggy.
 
-## Current solutions, their limitations, and APIs to fill the gaps
+## Current Solutions, Their Limitations, and APIs to Fill the Gaps
 
-This problem is generally tackled by systematically chunking and scheduling
-main thread work. Since long tasks and responsiveness are at odds, breaking up
-long tasks can help keep an app responisive when also *yielding to the browser's
-event loop*.
+Dealing with contention is largely a scheduling problem: to the degree that
+work can be reordered in an more optimal way, scheduling can have a positive
+impact. What makes this problem more pronounced on the web is that tasks run to
+completion, meaning the UA cannot interrupt a task to run high priority work
+like processing user input. This problem is generally tackled in userspace by
```

Comment:
```suggestion
impact. This problem is more pronounced on the web because tasks run to
completion: the UA cannot preempt a task to run higher priority work
like processing user input. This problem is generally tackled in userspace by
```

Dunno if "preempt" is actually better than "interrupt"; it's just the term I expect to see in this space.